beacond/cmd/main.go

```diff
@@ -44,44 +47,59 @@ func run() error {
 	// Build the node using the node-core.
 	nb := nodebuilder.New(
 		// Set the DepInject Configuration to the Default.
-		nodebuilder.WithDepInjectConfig[types.Node](
+		nodebuilder.WithDepInjectConfig[NodeT](
 			nodebuilder.DefaultDepInjectConfig()),
 		// Set the Runtime Components to the Default.
-		nodebuilder.WithComponents[types.Node](
+		nodebuilder.WithComponents[NodeT](
 			nodecomponents.DefaultComponentsWithStandardTypes(),
 		),
 	)
 
+	// Build the server using the server builder.
+	// sb := serverbuilder.New[NodeT, T, ValidatorUpdateT](
+	// 	serverbuilder.WithDepInjectConfig[NodeT, T, ValidatorUpdateT](
+	// 		nodebuilder.DefaultDepInjectConfig(),
+	// 	),
+	// 	serverbuilder.WithComponents[NodeT, T, ValidatorUpdateT](
+	// 		servercomponents.ProvideCometServer[NodeT, T, ValidatorUpdateT],
+	// 		nodecomponents.ProvideTxCodec[T],
+	// 	),
+	// )
+
 	// Build the root command using the builder
 	cb := clibuilder.New(
 		// Set the Name to the Default.
-		clibuilder.WithName[types.Node](nodebuilder.DefaultAppName),
+		clibuilder.WithName[NodeT](nodebuilder.DefaultAppName),
 		// Set the Description to the Default.
-		clibuilder.WithDescription[types.Node](nodebuilder.DefaultDescription),
+		clibuilder.WithDescription[NodeT](nodebuilder.DefaultDescription),
 		// Set the DepInject Configuration to the Default.
-		clibuilder.WithDepInjectConfig[types.Node](
+		clibuilder.WithDepInjectConfig[NodeT](
 			nodebuilder.DefaultDepInjectConfig(),
 		),
 		// Set the Runtime Components to the Default.
-		clibuilder.WithComponents[types.Node](
+		clibuilder.WithComponents[NodeT](
 			append(
 				clicomponents.DefaultClientComponents(),
 				// TODO: remove these, and eventually pull cfg and chainspec
 				// from built node
 				nodecomponents.ProvideNoopTxConfig,
 				nodecomponents.ProvideConfig,
 				nodecomponents.ProvideChainSpec,
-			),
+				nodecomponents.ProvideTxCodec[T],
+				servercomponents.ProvideCometServer[NodeT, T, ValidatorUpdateT],
```

Comment:
@archbear your depinject unexported error is because were sending a provider that accepts generics btw. I'm assuming this is because the depinject graph isn't built at runtime, so depinject won't be able to resolve the function type. 

so unfortunately we have to inject the hard types whenever we set the providers 🙁